app/src/main/java/org/mifos/selfserviceapp/presenters/LoanApplicationPresenter.java

```diff
@@ -89,7 +94,11 @@ public void onError(Throwable e) {
                     @Override
                     public void onNext(LoanTemplate loanTemplate) {
                         getMvpView().hideProgress();
-                        getMvpView().showLoanTemplateByProduct(loanTemplate);
+                        if (loanState == LoanState.CREATE) {
```

Comment:
@therajanmaurya In case of Loan Update,
We need to load the current value of loan purpose if it exists for the loan which is to be updated.
So `showUpdateLoanTemplateByProduct` load the spinner with all values from template and then check if it is being intialized for the first time and does it contains a loan purpose if then it loads those values otherwise it loads the values fetched from the template i.e. the principal amount ,currency and rest.